### DISCUSSION TITLE
pn532 rfid reader

### DISCUSSION THREAD

**Discussion by joreichhardt** [MiczFlor/RPi-Jukebox-RFID#1229] [Q&A] [answered]
Hi I'm getting an error that there's no pn532 defined reader connected

>Dec 23 13:05:43 phoniebox daemon_rfid_reader.py[8021]: Traceback (most recent call last):
>Dec 23 13:05:43 phoniebox daemon_rfid_reader.py[8021]: File "/home/pi/RPi-Jukebox-RFID/scripts/daemon_rfid_reader.py", line 19, in <module>
>Dec 23 13:05:43 phoniebox daemon_rfid_reader.py[8021]: reader = Reader()
>Dec 23 13:05:43 phoniebox daemon_rfid_reader.py[8021]: File "/home/pi/RPi-Jukebox-RFID/scripts/Reader.py", line 162, in init
>Dec 23 13:05:43 phoniebox daemon_rfid_reader.py[8021]: self.reader = Pn532Reader()
>Dec 23 13:05:43 phoniebox daemon_rfid_reader.py[8021]: File "/home/pi/RPi-Jukebox-RFID/scripts/Reader.py", line 135, in init
>Dec 23 13:05:43 phoniebox daemon_rfid_reader.py[8021]: pn532 = Pn532_i2c()
>Dec 23 13:05:43 phoniebox daemon_rfid_reader.py[8021]: NameError: name 'Pn532_i2c' is not defined
>Dec 23 13:05:43 phoniebox systemd[1]: phoniebox-rfid-reader.service: Main process exited, code=exited, status=1/FAILURE
>Dec 23 13:05:43 phoniebox systemd[1]: phoniebox-rfid-reader.service: Failed with result 'exit-code'.
>Dec 23 13:05:43 phoniebox systemd[1]: phoniebox-rfid-reader.service: Service RestartSec=100ms expired, scheduling restart.
>Dec 23 13:05:43 phoniebox systemd[1]: phoniebox-rfid-reader.service: Scheduled restart job, restart counter is at 5.
>Dec 23 13:05:43 phoniebox systemd[1]: Stopped Phoniebox RFID-Reader Service.
>Dec 23 13:05:43 phoniebox systemd[1]: phoniebox-rfid-reader.service: Start request repeated too quickly.
>Dec 23 13:05:43 phoniebox systemd[1]: phoniebox-rfid-reader.service: Failed with result 'exit-code'.
>Dec 23 13:05:43 phoniebox systemd[1]: Failed to start Phoniebox RFID-Reader Service.
`

after changing the name in deviceName.txt and back the log changes into 

>Dec 23 14:24:49 phoniebox systemd[1]: phoniebox-rfid-reader.service: Main process exited, code=exited, status=1/FAILURE
>Dec 23 14:24:49 phoniebox systemd[1]: phoniebox-rfid-reader.service: Failed with result 'exit-code'.
>Dec 23 14:24:50 phoniebox systemd[1]: phoniebox-rfid-reader.service: Service RestartSec=100ms expired, scheduling restart.
>Dec 23 14:24:50 phoniebox systemd[1]: phoniebox-rfid-reader.service: Scheduled restart job, restart counter is at 4.
>Dec 23 14:24:50 phoniebox systemd[1]: Stopped Phoniebox RFID-Reader Service.
>Dec 23 14:24:50 phoniebox systemd[1]: Started Phoniebox RFID-Reader Service.
>Dec 23 14:24:50 phoniebox daemon_rfid_reader.py[7495]: Could not find the device PN532
>Dec 23 14:24:50 phoniebox daemon_rfid_reader.py[7495]: .
>Dec 23 14:24:50 phoniebox daemon_rfid_reader.py[7495]:  Make sure it is connected
>Dec 23 14:24:50 phoniebox systemd[1]: phoniebox-rfid-reader.service: Main process exited, code=exited, status=1/FAILURE
>Dec 23 14:24:50 phoniebox systemd[1]: phoniebox-rfid-reader.service: Failed with result 'exit-code'.
>Dec 23 14:24:50 phoniebox systemd[1]: phoniebox-rfid-reader.service: Service RestartSec=100ms expired, scheduling restart.
>Dec 23 14:24:50 phoniebox systemd[1]: phoniebox-rfid-reader.service: Scheduled restart job, restart counter is at 5.
>Dec 23 14:24:50 phoniebox systemd[1]: Stopped Phoniebox RFID-Reader Service.
>Dec 23 14:24:50 phoniebox systemd[1]: phoniebox-rfid-reader.service: Start request repeated too quickly.
>Dec 23 14:24:50 phoniebox systemd[1]: phoniebox-rfid-reader.service: Failed with result 'exit-code'.
>Dec 23 14:24:50 phoniebox systemd[1]: Failed to start Phoniebox RFID-Reader Service.
`

Could not find device PN532

in fact it is there

>sudo i2cdetect -y 1
>sudo: unable to resolve host phoniebox: Name or service not known
>0 1 2 3 4 5 6 7 8 9 a b c d e f
>00: -- -- -- -- -- -- -- -- -- -- -- -- --
>10: -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- --
>20: -- -- -- -- 24 -- -- -- -- -- -- -- -- -- -- --
>30: -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- --
>40: -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- --
>50: -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- --
>60: -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- --
>70: -- -- -- -- -- -- -- --`

**Comment by s-martin** [comment]
Please execute `sudo python3 -m pip install --upgrade --force-reinstall -q -r ~/RPi-Jukebox-RFID/components/rfid-reader/PN532/requirements.txt`

**Reply by joreichhardt** [reply]
didn't help

**Reply by joreichhardt** [reply]
I did a fresh installation now and still have the issue

**Comment by joreichhardt** [comment, marked as answer]
Solution 

`sudo python3 -m pip install --upgrade --force-reinstall -q -r /home/pi/RPi-Jukebox-RFID/components/rfid-reader/RC522/requirements.txt`

**Reply by FleurGaBru** [reply]
Thanks for the fix! I was running into the same bug (?).
